mocca-client/src/main/java/com/paypal/mocca/client/MoccaFeignEncoder.java

```diff
@@ -114,33 +121,38 @@ static SelectionSet getSelectionSet(RequestTemplate requestTemplate) {
     }
 
     /**
-     * Returns an optional containing the operation variable annotation
-     * associated with a Feign request template object.
-     * If the request method doesn't have a variable annotation,
-     * an empty optional is returned.
+     * Returns a list containing the operation variables associated with a Feign request template object.
+     * Operation variables are the operation method parameters annotated with {@link com.paypal.mocca.client.annotation.Var}.
+     * If the request method doesn't have operation variables, an empty list is returned.
+     * The list is ordered according to {@code parameters} order.
      *
+     * @param parameters GraphQL operation method parameter values
      * @param requestTemplate the Feign request template object
-     * @return an optional containing the operation variable annotation
+     * @return a list containing the operation variables
      * associated with a Feign request template object
      */
-    static Optional<Variable> getVariable(RequestTemplate requestTemplate) {
-        Method method = requestTemplate.methodMetadata().method();
-        Parameter[] parameters = method.getParameters();
-        if (parameters.length == 0) {
-            return Optional.empty();
-        }
-        int varCount = 0;
-        int varIndex = 0;
+    private static List<Variable> getVariables(Object[] parameters, RequestTemplate requestTemplate) {
+        Parameter[] parametersMetadata = requestTemplate.methodMetadata().method().getParameters();
+        List<Variable> variables = new ArrayList<>(parameters.length);
         for (int i = 0; i < parameters.length; i++) {
-            if (parameters[i].getAnnotation(RequestHeaderParam.class) == null) {
-                varCount++;
-                varIndex = i;
+            Parameter parameterMetadata = parametersMetadata[i];
+            Var varAnnotation = parameterMetadata.getAnnotation(Var.class);
+            // FIXME It would be better if this check happened at client definition time, instead of request time
+            if (varAnnotation == null) {
+                if (parameterMetadata.getAnnotation(RequestHeaderParam.class) == null) {
```

Comment:
This "if" block appears to be copied and could be extract outside the if/else block I think.

Reply:
It is not, the condition is different.